text_generation/causal_lm/cpp/continuous_batching/library/src/continuous_batching_pipeline.cpp

```diff
@@ -232,11 +248,14 @@ class ContinuousBatchingPipeline::Impl {
         std::vector<GenerationResult> results;
         results.reserve(m_requests.size());
 
-        while (has_running_requests()) {
+        while (has_running_requests() && !out_of_memory()) {
             std::vector<GenerationResult> partial_results = step();
-            results.insert(results.end(), partial_results.begin(), partial_results.end());
+            if (partial_results.size() > 0)
```

Comment:
if in case of out of memory, we return sequence as IGNORED / OUT_OF_MEMORY, we always have partial_results at least having a single element.

Reply:
Removed this change, now we return partially generated sequences and partial_results are always not empty.